plugin/external/pluginEntry.go

```diff
@@ -125,7 +179,13 @@ func (e decodedExternalPluginEntry) toExternalPluginEntry(ctx context.Context, s
 
 	// If read content is static, it's likely it's not coming from a source that separately provides
 	// the size of that data. If not provided, update it since we know what it is.
-	if content, ok := methods["read"].result.(string); ok && !e.Attributes.HasSize() {
+	if sig := methods["read"]; sig.tupleValue != nil && !e.Attributes.HasSize() {
```

Comment:
It feels weird to have content validation here. Could we move that over to where the method tuples are being unmarshaled?  I think it's OK if this part just sets the size attribute.

Reply:
How can it get the correct size unless it knows the type?

Reply:
Ah, yeah I think it makes sense to move all the parsing to mungeToMethods.